azurestack/resource_arm_loadbalancer_nat_rule.go

```diff
@@ -0,0 +1,304 @@
+package azurestack
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2015-06-15/network"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurestack/azurestack/utils"
+)
+
+func resourceArmLoadBalancerNatRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLoadBalancerNatRuleCreateUpdate,
+		Read:   resourceArmLoadBalancerNatRuleRead,
+		Update: resourceArmLoadBalancerNatRuleCreateUpdate,
+		Delete: resourceArmLoadBalancerNatRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: loadBalancerSubResourceStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"protocol": {
+				Type:             schema.TypeString,
+				Required:         true,
+				StateFunc:        ignoreCaseStateFunc,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Tcp",
```

Comment:
Could we use the values provider by the SDK `string(network.TransportProtocolTCP),`